### PR TITLE
Save Defense Checkbox to not use FindChild every frame

### PR DESCRIPTION
**Pull request by ashaw596**
Save Defense Checkbox to not use FindChild every frame

### CODE REVIEW

**Comment by jgkamat** [comment]
Related to #674. May fix one of the (multiple) segfaults there.

Looks good at a first glance, I'm going to do another pass later on.

@jpfeltracco

**Comment by jpfeltracco** [comment]
I ran a couple of times on this branch and I haven't seen the find_child segfault and running out of memory didn't cause a segfault either so I'm cautiously optimistic about this.

I did get a new segfault which I posted to the original issue for now.

**Comment by jgkamat** [comment]
LGTM, @ashaw596, go ahead and merge when ready.

File: soccer/gameplay/GameplayModule.cpp [hunk 1]
```diff
@@ -341,32 +341,27 @@ void Gameplay::GameplayModule::run() {
...
                 if (bot && bot->visible) {
-                    theirBotVector->push_back(bot);
+                    theirBotVector.push_back(bot);
                 }
             }
             getMainModule().attr("set_their_robots")(theirBotVector);
 
             getMainModule().attr("set_game_state")(_state->gameState);
 
-            getMainModule().attr("set_system_state")(&_state);
```
Comment (jgkamat) [inline soccer/gameplay/GameplayModule.cpp]: Do you have any idea why this was a address before?

Comment (ashaw596) [inline soccer/gameplay/GameplayModule.cpp]: Really not sure since _state is an pointer already. sooooo....

**Pull request merged**